investpy/stocks.py

```
@@ -1506,3 +1506,176 @@ def search_stocks(by, value):
+        raise ValueError("ERR#0025: specified country value not valid.")
+
+    resource_package = 'investpy'
+    resource_path = '/'.join((('resources', 'stocks.csv')))
+    if pkg_resources.resource_exists(resource_package, resource_path):
+        stocks = pd.read_csv(pkg_resources.resource_filename(resource_package, resource_path), keep_default_na=False)
+    else:
+        raise FileNotFoundError("ERR#0056: stocks file not found or errored.")
+
+    if stocks is None:
+        raise IOError("ERR#0001: stocks object not found or unable to retrieve.")
+
+    if unidecode(country.lower()) not in get_stock_countries():
+        raise RuntimeError("ERR#0034: country " + country.lower() + " not found, check if it is correct.")
+
+    stocks = stocks[stocks['country'].str.lower() == unidecode(country.lower())]
+
+    stock = unidecode(stock.strip().lower())
+
+    if stock not in [unidecode(value.strip().lower()) for value in stocks['symbol'].tolist()]:
+        raise RuntimeError("ERR#0018: stock " + stock + " not found, check if it is correct.")
+
+    tag_ = stocks.loc[(stocks['symbol'].str.lower() == stock).idxmax(), 'tag']
+
+    headers = {
+        "User-Agent": random_user_agent(),
+        "X-Requested-With": "XMLHttpRequest",
+        "Accept": "text/html",
+        "Accept-Encoding": "gzip, deflate, br",
+        "Connection": "keep-alive",
+    }
+
+    url = 'https://www.investing.com/equities/' + str(tag_) + '-earnings'
+
+    req = requests.get(url=url, headers=headers)
+
+    if req.status_code != 200:
+        raise ConnectionError("ERR#0015: error " + str(req.status_code) + ", try again later.")
+
+    root_ = fromstring(req.text)
+    path_ = root_.xpath(".//table[contains(@id, 'earningsHistory')]")
+
+    if path_:
+        more_results_id = path_[0].get('id').replace('earningsHistory', '')
+
+        path_ = root_.xpath(".//table[@id='earningsHistory" + str(more_results_id) + "']/tbody/tr")
+
+        objs = list()
+
+        if path_:
```

Missing `else` statement, should raise an Exception just like all the other stock functions, please check https://github.com/alvarobartt/investpy/blob/master/investpy/stocks.py to see the latest changes and apply them here!

---

I think the else statement and Exception are already implemented as same as the following function.

https://github.com/alvarobartt/investpy/blob/16b14f0e69e1ce99d1e648ecc2cbfffe0fe39318/investpy/stocks.py#L803